### ISSUE TITLE
When using exponential as the activation function, the outputs of the CPU and GPU have large differences

### ISSUE THREAD

**Issue by PhyllisJi** [tensorflow/tensorflow#74791]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tf 2.12.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 20.04

### Python version

3.10

### Current behavior?

Input: [[25.05214, 6.4932823, 5.5203633, 12.618748, 27.186777, 3.7995481]]
CPU output: [[7.5858780e+10 6.6068842e+02 2.4972575e+02 3.0217081e+05 6.4130895e+11
  4.4680992e+01]]
GPU output: [[7.5858780e+10 6.6068842e+02 2.4972574e+02 3.0217081e+05 6.4130888e+11
  4.4680988e+01]]
Max distance: 65536.0

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

tf.random.set_seed(42)
tf.config.experimental.enable_op_determinism()

def chebyshev_distance(A: np.ndarray, B: np.ndarray):
    if A is None or B is None:
        return 0.0
    if A.shape != B.shape:
        return 9999999
    else:
        return float(np.max(np.abs(A - B)))


act_layer = tf.keras.layers.Activation(activation='exponential')
inp = np.array([[25.05214, 6.4932823, 5.5203633, 12.618748, 27.186777, 3.7995481]])
input_shape = inp.shape[1:]
act_layer.build(input_shape)


with tf.device('/CPU:0'):
    x_cpu = tf.constant(inp, dtype=tf.float32)
    output_cpu = act_layer(x_cpu)
    print("CPU output:", output_cpu.numpy())


if tf.config.list_physical_devices('GPU'):
    with tf.device('/GPU:0'):
        x_gpu = tf.constant(inp, dtype=tf.float32)
        output_gpu = act_layer(x_gpu)
        print("GPU output:", output_gpu.numpy())
else:
    print("GPU not available.")
    
output_diff = chebyshev_distance(output_cpu.numpy(), output_gpu.numpy())
print(output_diff)
```

**Comment by Venkat6871** [comment]
I tried to run your code on Colab using TF v2.17.0 with GPU and faced the same issue. Please find the [gist](https://colab.research.google.com/gist/Venkat6871/13adc3200c8bea5dd3f366d73c6b5616/74791_tf-2-17-0-v.ipynb) here for reference.
Thank you!

**Comment by PhyllisJi** [comment]
> I tried to run your code on Colab using TF v2.17.0 with GPU and faced the same issue. Please find the [gist](https://colab.research.google.com/gist/Venkat6871/13adc3200c8bea5dd3f366d73c6b5616/74791_tf-2-17-0-v.ipynb) here for reference. Thank you!

Yes, it's the same issue and the differences are more consistent.

**Comment by Venkat6871** [comment]
Hi @PhyllisJi ,
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) as this issue is more related to keras
Thank you!

**Comment by PhyllisJi** [comment]
> Hi @PhyllisJi , Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) as this issue is more related to keras Thank you!

I have posted ！Thanks

**Comment by Venkat6871** [comment]
Hi **@PhyllisJi** ,
Could you please close this issue since it is already being tracked in another repository?
Thank you!
